test/test_cuda.py

```diff
@@ -1093,6 +1094,23 @@ def test_var_stability(self):
         tensor = tensor.unsqueeze(1)
         self.assertEqual(tensor.var(0)[0], 0.03125)
 
+    def test_digamma(self):
+        def test(use_double=False):
+            cpu_tensor = torch.randn(10, 10, 10)
+            gpu_tensor = cpu_tensor.cuda()
+            zeros = torch.zeros(10, 10, 10)
+            if (use_double):
+                cpu_tensor = cpu_tensor.double()
+                gpu_tensor = gpu_tensor.double()
+                zeros = zeros.double()
+            cpu_out = cpu_tensor.digamma()
+            gpu_out = gpu_tensor.digamma()
+            norm_errors = (gpu_out - cpu_out.cuda()) / gpu_out
+            self.assertEqual(norm_errors, zeros)
+
+        test(True)
+        test(False)
```

Comment:
Quoting Richard from the PR on which this is based #3955 :
> The values of digamma on gpu when compared to cpu are the same to 6 digits. I tried making this a normal test and some output values were high (~100), resulting in an error of around 1e-3, which is above the error threshold

It's probably out of scope for this PR, but maybe you could change the generic tests to assert on relative error rather than absolute error?

Reply:
We have [means to control the required precision](https://github.com/probtorch/pytorch/blob/978b197bc8bbe44e1946367a763fd54c1ed55540/test/test_cuda.py#L326-L335), so it would be better to use that.

Reply:
meh it's easier to just implement a more accurate `digamma()` :wink:

Reply:
Ok I've set custom precisions as you suggest, however they're so lax (1e0 and 1e3) that I've kept @zou3519's the relative-error tests.